[Zhao Jinxin] iP

BuzzingBuzzard opened the pull request:
# Duke
> Due tomorrow? Do tomorrow. - Meme ([source](https://www.reddit.com/r/memes/comments/dhx7re/what_deadline/))

## Description
Duke frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. Let it manage your tasks for you :innocent:

And it is **FREE**!

## Features
- [x] Manage todos, deadlines, and events.
- [x] Find task by keyword
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the`<main>` method:
```java
public static void main(String[] args) {
    Duke duke = new Duke();
    duke.run();
}
```

Commit:
Add Gradle support

Commit:
Level-1

Commit:
Level-2

Commit:
Level 3

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5 Handle Errors

Commit:
Level-6 Delete

Commit:
Level-7 on branch

Commit:
Level-8 on branch

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
A-MoreOOP Use more OOP

Commit:
A-Packages Organize into Packages

Commit:
A-JUnit Add JUnit Tests

Commit:
A-JavaDoc on branch

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Change code by coding standards

Commit:
Add find function on branch

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Preshita01 commented on src/main/java/duke/DukeTaskList.java:
```diff
@@ -0,0 +1,178 @@
+            break;
+        case "D":
+            currTask = new Deadlines(bodySplit[0],
+                LocalDate.parse(bodySplit[1]),
```
Should the indentation for this wrapped line be 8 spaces more than the parent line instead? I think it is currently only 4 spaces.

I have noticed this in other parts of the code as well.

Preshita01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,39 @@
+package duke;
+
+import java.io.IOException;
+
```
Perhaps you want to add more descriptive header comments for this class and the public methods in it? I think the comments are missing for some other classes as well (e.g. DukeException and Ui classes)

Preshita01 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Method to read data from save file. If the save file and dir don't exist,
```
Maybe you can write the first sentence of the header comment as 'Reads data from save file.' so that it more closely aligns with the coding standards for the module?

This can also be applied to some of the other methods in the code.

Preshita01 reviewed (commented):
Overall the code is neat and easy to follow 👍 The in-line comments for some of the longer methods were definitely helpful (Sorry I misread the instructions and accidentally reviewed your code, I have removed the rest of my comments but I am not able to remove this. Sorry for the confusion!)

fullfatwasabi commented on src/main/java/duke/DukeTaskList.java:
```diff
@@ -0,0 +1,178 @@
+package duke;
+
+import duke.task.*;
```
Maybe should explicitly import here?

BenjaminLHR commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,124 @@
+                    throw new DukeException("Deadline must have a date and time written after a slash!\n");
+                }
+
+                String[] DdlDateTimeArr = slashSplitInput[1].split(" ", 3);
+                if (DdlDateTimeArr.length < 3) {
```
Perhaps you could abstract out 3 into a final variable in case the format of input changes.

BenjaminLHR commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,77 @@
+
+    public static void printDoneTask(String taskString) {
+        System.out.println(LINE + INDENT + "Nice! I've marked this task as done:\n"
+            + INDENT + INDENT + taskString + "\n"
+            + LINE);
```
Perhaps wrapped lines could be indented 8 spaces (instead of 4) more than the parent line.

BenjaminLHR commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,64 @@
+     * Constructor of the Events class.
+     *
+     * @param description description of an event in String.
+     * @param localDate date when the event happens.
+     *                  Must be in the format yyyy-mm-dd.
```
Should this be how multiline Javadoc comments are indented? I am not very certain myself.

BenjaminLHR reviewed (commented):
Just a few nits to fix. I like that your attributes and methods are sensibly named.

fullfatwasabi reviewed (commented):
Damn your code is really neat and well laid out. The only things I managed to find were the import thing and also indentation when wrapping perhaps.

Commit:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2122S1/ip

Commit:
Add Gradle support

Commit:
Add CheckStyle and correct styles.

Commit:
Add GUI

Commit:
Merge branch 'branch-Level-10'

Commit:
Add assertions

Add assertions to DukeTaskList and Storage classes.

Assertions document assumptions that should hold at their positions in code.

Commit:
Improve code quality

BuzzingBuzzard committed:
Merge pull request #1 from JinxinZhao315/branch-A-Assertions

Add assertions

Commit:
Merge branch 'branch-A-CodeQuality'

BuzzingBuzzard committed:
Merge pull request #2 from JinxinZhao315/branch-A-CodeQuality

Improve code quality

Commit:
Merge branch 'master' of https://github.com/JinxinZhao315/ip

Commit:
Add extension C-Update

Add the functionality to update description, date, time (for deadline), start_time and end_time (for events) of any given task.

Commit:
Refactor Parser class to improve readability

Commit:
Refactor code to improve readability

BuzzingBuzzard committed:
Update README.md

BuzzingBuzzard committed:
Update README.md

BuzzingBuzzard committed:
Update README.md

BuzzingBuzzard committed:
Update README.md

BuzzingBuzzard committed:
Update README.md

BuzzingBuzzard committed:
Set theme jekyll-theme-cayman

BuzzingBuzzard committed:
Update README.md

Commit:
Create better GUI

Add background colours to all GUI components.

BuzzingBuzzard committed:
Update README.md
